### DISCUSSION TITLE
Any Interest in a Mesh.GenerateCircle?

### DISCUSSION THREAD

**Discussion by jackdaus** [StereoKit/StereoKit#498] [Ideas]
Any interest in adding a method to generate a circular plane mesh?

I'm  in need of a circular plane mesh for my project. An easy workaround would be for me to just import a 3D model of a circular plane from a file. But if you think something like a `Mesh.GenerateCircle` would be a useful method to have around, I could contribute a PR!

**Comment by maluoi** [comment]
Yeah, I'll take a `Mesh.GenerateCircle`, that'd be awesome! I've thought about it a bit in the past, so I _do_ already have a bit of a wishlist:
- A parameter to specify "Forward" or "Facing" direction of the circle, as well as "Up" (see [GeneratePlane](https://stereokit.net/Pages/StereoKit/Mesh/GeneratePlane.html))
- A triangle layout that doesn't include a vertex at the center of the circle, [see the circle generator code in the Radial Hand menu](https://github.com/StereoKit/StereoKit/blob/develop/StereoKit/Framework/HandMenu/HandMenuRadial.cs#L370)
- A flag for two sided triangles? (I should totally add this to plane as well!)

And feel free to _not_ do those things if they get too tricky, I'll still appreciate any kind of circle generator :)

**Reply by jackdaus** [reply]
Cool! I'll start taking a look 😃

**Reply by jackdaus** [reply]
Alrighty so I got the feature pretty much all implemented! (https://github.com/jackdaus/StereoKit/tree/feature/mesh_generate_circle) I haven't yet implemented the two-sided triangle thing, but I'll take a look at doing that too (and maybe for the plane as well). Two questions whenever you get a chance:

- How should the UVs be mapped for the two sided version of the circle? Should the UVs be a mirror image on the flip side, or should the UVs be identical on both sides?
- I extended the existing Demo and Test files for the Mesh generators to include the new circle generator. But I couldn't figure out how to run the StereoKitDocumenter tool. I get a 404 when I try to find it on develop? (https://github.com/StereoKit/StereoKit/tree/develop/Tools/StereoKitDocumenter) I want to run the documenter locally to make sure I updated the Demo/Test files correctly.

**Reply by maluoi** [reply]
This looks fantastic! 

For double sided, I'd recommend just duplicate the vert and flip the normal, keep the UVs the same. :)

The documenter is actually moved out into [its own repository](https://github.com/StereoKit/StereoKit-Docs)! But to just run tests and generate screenshots, pass in the `-test` argument to StereoKitTest. There's actually a launch profile that does this for you:
![image](https://user-images.githubusercontent.com/8484884/197938194-e0b70b57-8fff-4930-99f5-f08a77ac0c83.png)

Now that I'm seeing the `subdivision` thing from an external perspective, I'm thinking it's probably a bit on the silly side! I think I'd love it if you could switch it either back to `spokes`, `vert_count` or maybe `resolution`? (feel free to come up with something better :D) And then clamp it to a minimum of 3. That feels like it'd be a lot more intuitive, and then we can move the others over to doing the same later on!

**Reply by jackdaus** [reply]
For sure, makes sense! Yeah I was kinda going back and forth between using "spokes" and "subdivisions" and settled on "subdivisions" just to match the other methods. But I think "spokes" is more precise and intuitive. I'll make the switch.
And thanks for pointing out the tests!
I'll try to get this wrapped up and make a PR soon when I have some time.
